devo/sender/data.py

```diff
@@ -625,3 +644,155 @@ def emit(self, record):
+            certificate_obj = crypto.load_certificate(
+                crypto.FILETYPE_PEM, certificate_raw)
+            certificates_chain = crypto.X509Store()
+            for certificate in pem.parse(chain_raw):
+                certificates_chain.add_cert(
+                    crypto.load_certificate(
+                        crypto.FILETYPE_PEM, str(certificate)))
+            store_ctx = crypto.X509StoreContext(
+                certificates_chain, certificate_obj)
+            store_ctx.verify_certificate()
+        except crypto.X509StoreContextError:
+            raise DevoSenderException(
+                "Error in config, the chain: " + config.chain +
+                " is not compatible with the certificate: " + config.cert)
+        return True
+
+    @staticmethod
+    def check_config_certificate_address(config):
+        """
+        Check if the certificate is compatible with the
+        address, also check is the address and port are
+        valid.
+
+        :param config -> Configuration object.
+        :return: Boolean true or raises an exception
+        """
+        sock = socket.socket()
+        context = SSL.Context(SSL.SSLv23_METHOD)
+        sock.settimeout(10)
+        connection = SSL.Connection(context, sock)
+        try:
+            connection.connect((config.address[0], config.address[1]))
+        except socket.timeout:
+            raise DevoSenderException(
+                "Error in config, the port: "
+                + str(config.address[1]) +
+                " is incorrect")
+        except ConnectionRefusedError:
+            raise DevoSenderException(
+                "Error in config, incorrect address: "
+                + str(config.address))
+        sock.setblocking(True)
+        connection.do_handshake()
+        server_certs = connection.get_peer_cert_chain()
+
+        chain = open(config.chain, "rb").read()
+        chain_certs = []
+        for _ca in pem.parse(chain):
+            chain_certs.append(
+                crypto.load_certificate(crypto.FILETYPE_PEM, str(_ca)))
+
+        try:
+            server_elements = {}
+            for key, value in server_certs[1].get_issuer().get_components():
+                server_elements[key.decode("utf-8")] = value
+
+            client_elements = {}
+            for key, value in chain_certs[0].get_subject().get_components():
+                client_elements[key.decode("utf-8")] = value
+
+            server_common_name = server_elements["CN"]
+            client_common_name = client_elements["CN"]
+
+            if server_common_name == client_common_name:
+                return True
+            else:
+                raise DevoSenderException(
+                    "Error in config, the chain: " + config.address +
+                    " is incorrect")
+        except Exception:
+            raise DevoSenderException(
+                "Error in config, the address: " + config.address[0] +
+                " is incorrect")
```

Comment:
I'm going to remove this `try` since it should not be needed, I was using it because I was using position arguments to get the `common name` from the certificate but the dictionary solved that problem.